scripts/gulpfiles/release_notes.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Gulp scripts for creating release notes.
+ */
+
+var gulp = require('gulp');
+var fetch = require('node-fetch');
+
+const baseUrl = "https://api.github.com/graphql";
+
+async function generateReleaseNotes(done) {
+  const goalDate = Date.parse('2021-03-01');
+  const githubToken = process.argv[4];
+  const headers = {
+    "Content-Type": "application/json",
+    Authorization: "bearer " + githubToken,
+  };
+
+
+  var note = `{Insert greeting},
+
+The {Insert Month and Year} release is out! In this release:
+
+{List major features in both core and samples}
+
+🎉 **Kudos** 🎉
+{Thank forums contributors and issue filers}
+
+**Third Party Plugins**
+{List third party plugins that have been released in the last quarter}
+
+**Core contributions**
+{Reorganize into thank-yous and check for missed people}
+${await getContributors('blockly', 'develop', goalDate, headers)}
+
+**Samples contributions**
+{Reorganize into thank-yous and check for missed people}
+${await getContributors('blockly-samples', 'master', goalDate, headers)}
+
+🔨 **Breaking Changes in core** 🔨 
+${await getBreakingChanges(headers)}
+
+Full release notes are below. As always, we welcome bug reports and pull requests!
+
+Cheers,
+{Your Name}
+
+**Issues Closed**:
+${await getClosedIssues(headers)}
+
+**Pull Requests**:
+${await getMergedPrs(headers)}`;
+
+  console.log(note);
+
+  done();
+}
+
+async function getContributors(repo, baseRef, goalDate, headers) {
+  const collaborators = await getCollaborators(repo, headers);
+
+  let cursor = '';
+  const contributors = new Map();
+  do {
+    try {
+      const response = await fetch(baseUrl, {
+        method: "POST",
+        headers: headers,
+        body: makeContributorQuery(repo, baseRef, cursor),
+      });
+      const json = await response.json();
+      const pullRequestData = json.data.repository.pullRequests;
+      const pulls = pullRequestData.nodes;
+      cursor = pullRequestData.pageInfo.endCursor;
+    
+      for (let i = 0; i < pulls.length; i++) {
+        const pull = pulls[i];
+        const createdDate = Date.parse(pull.createdAt);
+        if (createdDate < goalDate) {
+          cursor = '';
+          break;
+        }
+
+        if (collaborators.has(pull.author.login)) {
+          continue;
+        }
+        let contributions = contributors.get(pull.author.login);
+        if (!contributions) {
+          contributions = [];
+          contributors.set(pull.author.login, contributions);
+        }
+        contributions.push({
+          title: pull.title,
+          url: pull.url,
+          number: pull.number
+        })
+      }
+
+    } catch (error) {
+        console.log(error);
```

Comment:
Indentation: should be +2, not +4.  (Maybe just run `clang-format` on the whole file?)